tools/pre-commit.sh

```diff
@@ -1,28 +1,36 @@
 #!/usr/bin/env bash
+# Copy or link this script into .git/hooks/
+#   $ cp tools/pre-commit.sh .git/hooks/pre-commit
+# It runs automatically in the project root directory (parent of .git/).
 
-python -m black --check ./ > /dev/null 2>&1
-need_format=$?
-set -e
-if [ $need_format -ne 0 ]
-then
-    python -m black ./
-    echo Some Python files were formatted
-    echo You need to do git add and git commit again
-    exit $need_format
+EXIT_STATUS=0
+PROJECT_ROOT=`git rev-parse --show-toplevel`
+
+# Environmental variable used to notify hooks we are running in a workflow.
+ARMORY_CI_TEST="${ARMORY_CI_TEST:-0}"
+
+# Collect tracked files based on ARMORY_CI_TEST
+TRACKED_FILES="git diff HEAD --name-only"
+if [ "${ARMORY_CI_TEST}" -ne 0 ]; then
+    TRACKED_FILES="git ls-files"
 fi
-set +e
-
-python -m tools.format_json --check > /dev/null 2>&1
-need_format=$?
-set -e
-if [ $need_format -ne 0 ]
-then
-    python -m tools.format_json
-    echo Some JSON files were formatted
-    echo You need to do git add and git commit again
-    exit $need_format
+
+
+pushd $PROJECT_ROOT > /dev/null
+    # Source hooks
+    for FILE in `ls -1 ./tools/hooks/*.sh | sort`; do
+        echo "Importing ${FILE}..."
+        source "${FILE}"
+        if [ $? -ne 0 ]; then
+            EXIT_STATUS=1
+        fi
+    done
+popd > /dev/null
```

Comment:
This way of structuring the calls to the hooks is really hard to understand and harder to debug. A number of things make this hard:
1) Setting of environment variables that are then used in other bash script hooks.
2) The hook bash scripts are not executable, and there is not a clear interface to use them without invoking the main pre-commit hook, which makes testing them difficult.

Instead, can you give them a clear API so that they can be run separately? And please document the interface.

Alternatively, let's make full use of `source` - you can just execute TRACKED_FILES in the pre-commit hook and the other scripts can use that directly. Also, `pushd` and `popd` wouldn't be needed in the hooks, as you have already done that in the pre-commit.